### PR TITLE
Faster evaluation for MPM2

### PR DESCRIPTION
**Pull request by schaepermeier**
This PR adds support for extracting all instance-specific information for MPM2 problems from the original Python environment, and subsequently evaluating the problems within R instead of Python. This behavior is controlled by an additional parameter `evaluation.env`, which per default is set to the faster R.
It can be significantly faster in multi-objective optimization settings as the Python interface always only stores the configuration of the latest MPM2 function called, leading to expensive reconfigurations for each batch of evaluations.
Even in single-objective settings it is somewhat faster (about 20% for me).

In multi-objective settings it can easily be orders of magnitude faster when the MPM2 instance generation algorithm takes comparatively long to configure the problem. For example, on my machine:

```r
> env = "Python"
> 
> fn1 = makeMPM2Function(16, 2, "random", 8, evaluation.env = env)
> fn2 = makeMPM2Function(16, 2, "random", 9, evaluation.env = env)
> 
> system.time({
+   sapply(1:1e2, function(i) {
+     fn1(c(0.5, 0.5))
+     fn2(c(0.5, 0.5))
+   })
+ })
   user  system elapsed 
  1.239   0.002   0.613 
> library(smoof)
> 
> env = "R"
> 
> fn1 = makeMPM2Function(16, 2, "random", 8, evaluation.env = env)
> fn2 = makeMPM2Function(16, 2, "random", 9, evaluation.env = env)
> 
> system.time({
+   sapply(1:1e2, function(i) {
+     fn1(c(0.5, 0.5))
+     fn2(c(0.5, 0.5))
+   })
+ })
   user  system elapsed 
  0.018   0.000   0.011 
> 
```

Tests to check matching outputs are also provided. Let me know if you need me to change anything!

Cheers,
Lennart

### CODE REVIEW

**Review by jakobbossek — APPROVED** [review]
Looks good!

**Comment by jakobbossek** [comment]
There are some conflicts with the NEWS files most likely because I just merged your PR on the extended biobj. BBOB suite. Please fix ASAP.

**Comment by schaepermeier** [comment]
That should fix it :-)

File: R/sof.mpm2.R [hunk 1]
```diff
@@ -12,6 +12,12 @@
 #'   of elliptically shaped peaks.
 #' @param peak.shape [\code{character(1)}]\cr
 #'   Shape of peak(s). Possible values are \dQuote{ellipse} and \dQuote{sphere}.
+#' @param evaluation.env [\code{character(1)}]\cr
+#'   Evaluation environment after the function was created. Possible values are
+#'   \dQuote{R} (default) and \dQuote{Python}. The original generation of the
```
Comment (jakobbossek) [inline R/sof.mpm2.R:17]: Hmm, the case-sensitivity here is actually not user-friendly. Should we allow arbitrary string? E.g., r or R or pYthon or PyThOn?

File: NEWS [hunk 2]
```diff
@@ -4,6 +4,8 @@ smoof 1.6.0.4
 * Fixed bug in getLoggedValues when logging of x-values was set to FALSE in addLoggingWrapper
```
Comment (jakobbossek) [inline NEWS:4]: General comment: given your nice contribution I suggest to add yourself to the list of contributors in DESCRIPTION 🥳

**Review by jakobbossek — CHANGES_REQUESTED** [review]

**Review by jakobbossek — APPROVED** [review]
Great!

File: R/sof.mpm2.R (hunk 1 above)
Comment (schaepermeier) [reply R/sof.mpm2.R]: Good catch, I made it case-insensitive now

**Comment by jakobbossek** [comment]
Thanks Lennart for your super-fast responses! Merging 😎

**Pull request merged**